src/instructlab/chat/chat.py

```diff
@@ -391,6 +392,10 @@ def start_prompt(self, logger, content=None, box=True):
                             )
                             self.info["messages"].pop()
```

Comment:
this is trimming the newest message and also recent llama-cpp versions don't seem to tolerate changing the message list size

Reply:
Ignore the confusion about "newest" vs "latest" -- I read a comment out-of-place.

The main thing here though is that we still get InternalError happening when we try to shorten messages this way and I think we're going to make that a separate issue.